### PR TITLE
build: handle sanitizers in cmake

### PR DESCRIPTION
**Pull request by kinnou02**

### CODE REVIEW

File: source/cmake_modules/CompilationFlags.cmake [hunk 1]
```diff
@@ -51,3 +51,16 @@ endif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
 
 # Debug flags
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -DDEBUG=1 -fno-omit-frame-pointer")
+
+set(NAVITIA_ALLOCATOR "tcmalloc")
+
+option(USE_SANATIZER "build with the desired sanitizer enabled. usual values: address, thread, leak, undefined" OFF)
```
Comment (pbougue) [inline source/cmake_modules/CompilationFlags.cmake]: s/SANATIZER/SAN`I`TIZER/g ;)

**Review by pbougue — COMMENTED** [review]
:+1: Just the rename for later grep.

File: source/cmake_modules/CompilationFlags.cmake [hunk 2]
```diff
@@ -51,3 +51,16 @@ endif("${CMAKE_CXX_COMPILER_ID}" MATCHES "Clang")
...
+if(USE_SANATIZER)
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fsanitize=${USE_SANATIZER}")
```
Comment (pbougue) [inline source/cmake_modules/CompilationFlags.cmake]: You should also flag `-fno-omit-frame-pointer`

Comment (benoit-bst) [reply source/cmake_modules/CompilationFlags.cmake]: I see the same about the flag `-fno-omit-frame-pointer`

File: source/cmake_modules/CompilationFlags.cmake (hunk 1 above)
Comment (kinnou02) [reply source/cmake_modules/CompilationFlags.cmake]: `s/SAN\zsA\zeTIZER/I/g` :D

**Comment by kinnou02** [comment]
Jenkins, test this please.

**Review by pbougue — APPROVED** [review]

**Review by benoit-bst — APPROVED** [review]
Tested during my research about the Seg fault.

**Review by nonifier — APPROVED** [review]

**Pull request merged**